util/election2/etcd/election.go

```diff
@@ -59,8 +61,9 @@ func (e *Election) WithMastership(ctx context.Context) (context.Context, error)
 		cancel()
 		return nil, ctx.Err()
 	case rsp, ok := <-ch:
-		if !ok || rsp.Kvs[0].CreateRevision != etcdRev {
-			// Mastership has been overtaken in the meantime, or not capturead at all.
+		kv := rsp.Kvs[0]
+		if !ok || kv.CreateRevision != etcdRev || string(kv.Value) == resignID {
```

Comment:
I might be missing something, but the last two conditions here
`kv.CreateRevision != etcdRev` and `kv.Value == resignID` might be redundant.
If kv.Value changes (eg. to resign), would `etcdRev` get updated?

Reply:
`CreateRevision` doesn't change as long as the master keeps mastership. It's the revision of when the `etcd` key was **first** created.

Reply:
Got it, thanks.